common/src/main/java/tachyon/LeaderSelectorClient.java

```diff
@@ -62,11 +62,18 @@ public LeaderSelectorClient(String zookeeperAddress, String electionPath, String
 
     // Create a leader selector using the given path for management.
     // All participants in a given leader selection must use the same path.
-    // ExampleClient here is also a LeaderSelectorListener but this isn't required.
     CuratorFramework client =
         CuratorFrameworkFactory.newClient(mZookeeperAddress, new ExponentialBackoffRetry(
             Constants.SECOND_MS, 3));
     client.start();
+
+    // Sometimes, if the master crashes and restarts too quickly, zookeeper thinks the new client is
+    // still an old one. Close the "old" client and recreate a new one.
+    client.close();
```

Comment:
I see. I'm not sure if zk has more elegant ways to get around this, but extracting this into a function with detailed explanations would be very helpful.